field/TinyLang/Field/Typed/Parser.hs

```diff
@@ -53,3 +57,31 @@ parseExpr
     :: forall f m. (MonadError String m, MonadSupply m, TextField f)
     => String -> m (SomeUniExpr f)
 parseExpr = fmap _scopedValue . parseScopedExpr
+
+
+parseScopedProgram'
```

Comment:
Let's call it `parseScopedProgramFrom` or something, I don't like `'` in top-level names (unless they're about strictness and apart from a few exceptions like `Lens'`).